Feature/invitations/invitations component

Gzort opened the pull request:
Continuation of Invitations develop

Gzort committed:
Merge remote-tracking branch 'origin/feature/auth/LoginComponent' into feature/main/home

Gzort committed:
Add HomeUI for student and teacher
Add MainUI
Import required icons into drawable

Gzort committed:
Fixed warnings

Commit:
Renamed UseRole -> UserRole

Gzort committed:
Create module Home

Commit:
Now userId is saved after auth

Gzort committed:
Merge branch 'refs/heads/feature/profile/details' into feature/main/home

# Conflicts:
#	features/auth/src/main/java/ru/itmo/edugoolda/features/auth/presentation/register/RealRegisterComponent.kt
#	features/auth/src/main/java/ru/itmo/edugoolda/features/auth/presentation/register/RegisterUI.kt
#	settings.gradle.kts

Gzort committed:
Add Invitations Data

Gzort committed:
Add Invitations UI

Gzort committed:
Add UserInfo DTO

Gzort committed:
Add InvitationRepository with implementation
Add DI in features/invitations
Add DI in data/invitations

Gzort committed:
Detekt fixes

Gzort commented:
Я знаю, что здесь остались ошибки по HomeUI. Их я в соответствующей ветке исправлю

Krakozaybr commented on data/invitations/src/main/java/ru/itmo/edugoolda/data/invitations/api/InvitationRepository.kt:
```diff
@@ -0,0 +1,5 @@
+package ru.itmo.edugoolda.data.invitations.api
+
+interface InvitationRepository {
+    suspend fun action(invitationId: String, action: String)
```
Заведи для `invitationId` отдельный `value class`, а для `action` - отдельный `enum`

Давай назовём метод `respondToInvitation`

Krakozaybr commented on data/invitations/src/main/java/ru/itmo/edugoolda/data/invitations/internal/InvitationsApi.kt:
```diff
@@ -0,0 +1,13 @@
+import de.jensklingenberg.ktorfit.http.Query
+import ru.itmo.edugoolda.data.invitations.internal.dto.InvitationListResponse
+
+interface InvitationsApi {
```
У тебя есть `InvitationsApi` и `InvitationApi`, зачем сразу два - непонятно, давай сделаем один интерфейс

Krakozaybr commented on data/invitations/src/main/java/ru/itmo/edugoolda/data/invitations/internal/InvitationApi.kt:
```diff
@@ -0,0 +1,11 @@
+
+interface InvitationApi {
+    @POST("/api/v1/join_request/{requestUuid}")
+    suspend fun action(@Path("requestUuid") requestId: String, @Body request: InvitationActionRequest, )
```
Давай назовём `respondToJoinRequest`

Krakozaybr commented on data/user/src/main/java/ru/itmo/edugoolda/data/user/api/UserInfo.kt:
```diff
@@ -0,0 +1,14 @@
+import kotlinx.serialization.Serializable
+
+data class UserInfo(
+    val id: String,
```
Давай тип будет `UserId`

Krakozaybr commented on data/user/src/main/java/ru/itmo/edugoolda/data/user/api/UserInfo.kt:
```diff
@@ -0,0 +1,14 @@
+    val id: String,
+    val name: String,
+    val email: String,
+    val role: String,
```
Давай тип будет `UserRole`

Krakozaybr commented on data/user/src/main/java/ru/itmo/edugoolda/data/user/api/internal/dto/UserInfoDTO.kt:
```diff
@@ -0,0 +1,3 @@
+package ru.itmo.edugoolda.data.user.api.internal.dto
+
+class UserInfoDTO
```
Нигде не используется, удаляй

Krakozaybr commented on features/invitations/src/main/java/ru/itmo/edugoolda/features/invitations/presentation/InvitationsUI.kt:
```diff
@@ -0,0 +1,54 @@
+                    studentName = it.sender.name,
+                    date = it.date,
+                    onAcceptClick = { component.onAcceptClick(it) },
+                    onDeclineClick = { component.onAcceptClick(it) }
```
Должен вызываться `onDeclineClick`

Gzort committed:
Renamed Invitations<sth> to JoinRequest<sth>
Renamed modules invitations to join-requests
Add value class JoinRequestId and enum class JoinRequestAction
Merge InvitationsApi into InvitationApi
Delete unused UserInfoDTO
Changed role type to UserRole in UserInfo

Gzort committed:
Delete unused imports

Krakozaybr commented on data/join-requests/src/main/java/ru/itmo/edugoolda/data/invitations/internal/JoinRequestRepositoryImpl.kt:
```diff
@@ -0,0 +1,23 @@
+        joinRequestId: JoinRequestId,
+        action: JoinRequestAction
+    ) {
+        val joinRequestResponseRequest = when (action) {
+            JoinRequestAction.Accept -> JoinRequestResponseRequest("accept")
+            JoinRequestAction.Decline -> JoinRequestResponseRequest("decline")
+            JoinRequestAction.DeclineAndBan -> JoinRequestResponseRequest("decline_and_ban")
+        }
```
Лучше это делать немножко по-другому:
```kotlin
val request = JoinRequestResponseRequest(
    action = when (action) {
        JoinRequestAction.Accept -> "accept"
        JoinRequestAction.Decline -> "decline"
        JoinRequestAction.DeclineAndBan -> "decline_and_ban"
    }
)
```

Krakozaybr commented on data/join-requests/src/main/java/ru/itmo/edugoolda/data/invitations/api/JoinRequestRepository.kt:
```diff
@@ -0,0 +1,5 @@
+package ru.itmo.edugoolda.data.invitations.api
+
+interface JoinRequestRepository {
+    suspend fun respondToJoinRequest(joinRequestId: JoinRequestId, action: JoinRequestAction)
+}
```
Объедини `JoinRequestRepository` и `JoinRequestListRepository`, они об одном, в разделении нет особого смысла

Gzort committed:
Merge JoinRequestListRepository into JoinRequestRepository

Gzort committed:
Merge branch 'develop' into feature/invitations/InvitationsComponent
